packages/promotion/src/services/promotion-module.ts

```diff
@@ -231,6 +231,15 @@ export default class PromotionModuleService<
       PromotionTypes.ComputeActionAdjustmentLine
     >()
     const methodIdPromoValueMap = new Map<string, number>()
+    const automaticPromotions = preventAutoPromotions
+      ? []
+      : await this.list({ is_automatic: true }, { select: ["code"] })
```

Comment:
thought: we might want to do something more clever than this at some point - imagine a store with thousands of automatic discounts. This would introduce a lot of overhead.

Reply:
Yes, agree. I'll create a ticket for revisiting this.